dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -196,4 +140,65 @@ private string DebuggerDisplay
             return display;
         }
     }
+
+    #region Error handling
+    /// <summary>
+    /// Whether an error occurred while executing functions in the pipeline.
+    /// </summary>
+    public bool ErrorOccurred { get; private set; }
+
+    /// <summary>
+    /// Error details.
+    /// </summary>
+    public string LastErrorDescription { get; private set; } = string.Empty;
+
+    /// <summary>
+    /// When an error occurs, this is the most recent exception.
+    /// </summary>
+    public Exception? LastException { get; private set; }
+
+    /// <summary>
+    /// Call this method to signal when an error occurs.
+    /// In the usual scenarios this is also how execution is stopped, e.g. to inform the user or take necessary steps.
+    /// </summary>
+    /// <param name="errorDescription">Error description</param>
+    /// <param name="exception">If available, the exception occurred</param>
+    /// <returns>The current instance</returns>
+    public SKContext Fail(string errorDescription, Exception? exception = null)
+    {
+        this.ErrorOccurred = true;
+        this.LastErrorDescription = errorDescription;
+        this.LastException = exception;
+        return this;
+    }
+    #endregion
+
+    #region Obsolete
+    /// <summary>
+    /// Shortcut into user data, access variables by name
+    /// </summary>
+    /// <param name="name">Variable name</param>
+    [Obsolete("Use SKContext.Variables instead. The SKContext[...] indexer will be removed in a future release.")]
+    public string this[string name]
+    {
+        get => this.Variables[name];
+        set => this.Variables[name] = value;
+    }
+
+    /// <summary>
+    /// The token to monitor for cancellation requests.
+    /// </summary>
+    [Obsolete("Add a CancellationToken param to SKFunction method signatures instead of retrieving it from SKContext.")]
+    public CancellationToken CancellationToken { get; } = default;
+
+    /// <summary>
+    /// Semantic memory
+    /// </summary>
+    [Obsolete("Memory no longer passed through SKContext. Instead, initialize your skill class with the memory provider it needs.")]
```

Comment:
It would be nice to reference the kernel syntax sample that shows the new pattern so that there is no confusion about what change needs to be made.

Reply:
This Memory obsoleted thing is already in and released (this PR just moves the method within the file). But your point stands... should point at new use pattern where applicable.
In this case, I don't think we have any examples other than the Planners, of functions using Memory. That's one of the reasons this deprecation was a fairly easy choice to make; I don't believe this access pattern has much usage.